AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+
+    // Holds all the tasks currently shown on the task list.
+    val tasks = remember { mutableStateListOf(*allTasks) }
+
+    // Holds the topic that is currently expanded to show its body.
+    var expandedTopic by remember { mutableStateOf<String?>(null) }
+
+    // True if the message about the edit feature is shown.
+    var editMessageShown by remember { mutableStateOf(false) }
+
+    // Starts loading weather data. This takes 3 seconds.
+    suspend fun loadWeather() {
+        if (!weatherLoading) {
+            weatherLoading = true
+            delay(3000L)
+            weatherLoading = false
+        }
+    }
+
+    // Shows the message about edit feature.
+    suspend fun showEditMessage() {
+        if (!editMessageShown) {
+            editMessageShown = true
+            delay(3000L)
+            editMessageShown = false
+        }
+    }
+
+    LaunchedEffect(Unit) {
+        loadWeather()
+    }
+    val lazyListState = rememberLazyListState()
+
+    // The background color. The value is changed by the current tab.
+    val backgroundColor by animateColorAsState(if (currentTab == 0) purple100 else green300)
+
+    // The coroutine scope for event handlers calling suspend functions.
+    val scope = rememberCoroutineScope()
+    Scaffold(
+        topBar = {
+            HomeTabBar(
+                backgroundColor = backgroundColor,
+                currentTab = currentTab,
+                onTabSelected = { currentTab = it }
+            )
+        },
+        backgroundColor = backgroundColor,
+        floatingActionButton = {
+            HomeFloatingActionButton(
+                extended = lazyListState.isScrollingUp(),
+                onClick = {
+                    scope.launch {
+                        showEditMessage()
+                    }
+                }
+            )
+        }
+    ) {
+        LazyColumn(
+            contentPadding = PaddingValues(start = 16.dp, top = 32.dp, end = 16.dp, bottom = 32.dp),
+            state = lazyListState
+        ) {
+            // Weather
+            item { Header(title = stringResource(R.string.weather)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            item {
+                Surface(
+                    modifier = Modifier.fillMaxWidth(),
+                    elevation = 2.dp
+                ) {
+                    if (weatherLoading) {
+                        LoadingRow()
+                    } else {
+                        WeatherRow(onRefresh = {
+                            scope.launch {
+                                loadWeather()
+                            }
+                        })
+                    }
+                }
+            }
+            item { Spacer(modifier = Modifier.height(32.dp)) }
+
+            // Topics
+            item { Header(title = stringResource(R.string.topics)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            items(count = allTopics.size) { i ->
```

Comment:
`items(allTopics)`

Reply:
Make sure to import `LazyListScope.items` otherwise this won't work (import isn't suggested)

Reply:
Done